From 2a6500ceb96ebd7d856630a6a920f3898c765b1f Mon Sep 17 00:00:00 2001
From: GUO Zihua <guozihua@huawei.com>
Date: Thu, 3 Nov 2022 10:52:02 +0800
Subject: [PATCH 541/595] [Huawei] pgp: Check result of crypto_alloc_shash
 properly

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187945

--------------------------------

Return of crypto_alloc_shash would be either a valid pointer or an error
pointer. We should check the result with IS_ERR.

Signed-off-by: GUO Zihua <guozihua@huawei.com>
---
 crypto/asymmetric_keys/pgp_public_key.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crypto/asymmetric_keys/pgp_public_key.c b/crypto/asymmetric_keys/pgp_public_key.c
index 928029a13435..e49bb79736da 100644
--- a/crypto/asymmetric_keys/pgp_public_key.c
+++ b/crypto/asymmetric_keys/pgp_public_key.c
@@ -131,7 +131,7 @@ static int pgp_generate_fingerprint(struct pgp_key_data_parse_context *ctx,
 	ret = -ENOMEM;
 	tfm = crypto_alloc_shash(pgp->version < PGP_KEY_VERSION_4 ?
 				 "md5" : "sha1", 0, 0);
-	if (!tfm)
+	if (IS_ERR(tfm))
 		goto cleanup;
 
 	digest = kmalloc(sizeof(*digest) + crypto_shash_descsize(tfm),
-- 
2.17.1

